x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -140,11 +140,164 @@ func (k Keeper) ComputeMinPowerToOptIn(ctx sdk.Context, chainID string, bondedVa
 	return 0
 }
 
-// ShouldConsiderOnlyOptIn returns true if `validator` is opted in, in `chainID.
-func (k Keeper) ShouldConsiderOnlyOptIn(ctx sdk.Context, chainID string, validator stakingtypes.Validator) bool {
-	consAddr, err := validator.GetConsAddr()
-	if err != nil {
-		return false
+// CapValidatorSet caps the provided `validators` if chain `chainID` is an Opt In chain with a validator-set cap. If cap
+// is `k`, `CapValidatorSet` returns the first `k` validators from `validators` with the highest power.
+func (k Keeper) CapValidatorSet(ctx sdk.Context, chainID string, validators []types.ConsumerValidator) []types.ConsumerValidator {
+	if topN, found := k.GetTopN(ctx, chainID); found && topN > 0 {
+		// is a no-op if the chain is a Top N chain
```

Comment:
Why do we not allow this? TopN of 66% is only 24 validators on the hub right now.
Could easily imagine TopN=66 with ValSetCap=50 or 100 to make sense; but maybe it's not worth the effort at this point, since we will likely have all consumers come in as opt-in for now